### PR TITLE
Add support for more bibtex types 

### PR DESCRIPTION
**Pull request by arichardson**
This adds `@thesis`,`@report` and `@conference` as well as tests for the type field.

### CODE REVIEW

**Comment by gcushen** [comment]
Thanks for your contribution.

#68 has just been merged and there are now Git conflicts with your PR.

Please can you consider taking at the changes made in #68 and refactoring?

**Comment by arichardson** [comment]
Sure will do shortly. Looks like I made similar frontmatter changes (using toml instead of yaml) in my local changes: https://github.com/arichardson/academic-admin/tree/my-changes

I'll also try and rebase those other changes.

**Comment by arichardson** [comment]
Rebased now. Will add another PR with my other changes shortly.

**Comment by gcushen** [comment]
@arichardson please can you confirm if this PR works without a Docker compose file present?

There are reports in #69 of a possible bug introduced by the recently merged PR #68 ...

**Comment by arichardson** [comment]
@gcushen yes, this PR works but I encountered the same bug and it should be fixed by https://github.com/arichardson/academic-admin/commit/ff113084bcf4f78aa7b0dba28e92b43dd2323a78 which I haven't pushed as a pr yet. Maybe you could cherry-pick that commit to fix the docker issue?

File: academic/editFM.py [hunk 1]
```diff
@@ -20,7 +20,10 @@ def load(self, file: Path):
         self.content = []
         self.path = self.base_path / file
 
-        file = open(self.path, "r").readlines()
+        if not self.path.exists():
+            self.fm = dict()
+            return
+        file = self.path.open("r").readlines()
```
Comment (gcushen) [inline academic/editFM.py:26]: I can't see where we are closing the file we opened. Perhaps we should explicitly close it after the contents has been processed?

Also, let's read the file with UTF-8 encoding, just as we've done for the file writing in #70?

Might be worth rebasing at the same time (fortunately, should be pretty automatic this time).

**Review by gcushen — CHANGES_REQUESTED** [review]

File: academic/editFM.py (hunk 1 above)
Comment (arichardson) [reply academic/editFM.py:26]: That bug already existed, but I'll open a new PR.

**Pull request closed without merging**